### PR TITLE
Overhaul all active front end pages with the V1 Design Assets 

### LINKED ISSUE #69: Update Start Game Page to use new design assets

### LINKED ISSUE #86: Overhaul all active front end pages with the V1 Design Assets

### PR DESCRIPTION
**Pull request by dzx11**
Closes #86 

Routes affected:
/* (landing page)
/start-game
/join-game
/game-info/:gameId
/challenge-submission/:phone/:game

I took some liberties with the design and there's definitely still room for improvement but hopefully this is good enough for an MVP. I couldn't get the "COVID-19" text to appear behind the "START A GAME" button look in the UX Prototype in a reasonable amount of type so I skipped this. I think the best way to add it would be to get an image of the text and put it behind the button, similarly to how I did the VIR-US logo. Maybe we could get some help from the designers on this?

Also just making a note here, if you make changes on the FE please make sure that it looks OK on most popular devices/resolutions. Going forward we shouldn't be merging any code that breaks the layout in any popular device (phones, tablets). I think I checked most of the devices in this pull but let me know if you notice anything off.
@un7c0rn  @bernardbaker

### CODE REVIEW

**Review by un7c0rn — DISMISSED** [review]
Awesome. Definitely coming together nicely! I also agree that as we get closer to MVP it's a good idea for us to each test out the UI changes prior to merge. This will help prevent regressions, as automated web UI testing seems out of scope for the short term.

Some minor notes:

1. Can we try with updated VIR-US logo?

2. I think shrinking the inverted triangle vector will allow us to lift the "start game" button higher on the page so that it's more visible to users.

3. I think the "start game" / "join game" button can just be a big button but does not need to span the horizontal length of the page. I see what the designer was going for but it makes it harder to see / use.

4. Note for a separate PR: Is it feasible to animate the countdown so that the numbers are constantly changing?

Thanks!
Brandon

**Comment by bernardbaker** [comment]
Hi everyone,

I agree it's coming together. I'll look through the code and provide some
feedback.

Well done David. I respect the effort you're putting in.

Regards,

Bernard T. A. Baker

File: react_front_end/src/pages/LandingPage.js [hunk 1]
```diff
@@ -1,15 +1,17 @@
-import { withStyles } from "@material-ui/core";
-import { makeStyles } from "@material-ui/core/styles";
 import React, { useContext, useEffect, useState } from "react";
```
Comment (bernardbaker) [inline react_front_end/src/pages/LandingPage.js:1]: Could you remove the unused variables from this file?

File: react_front_end/src/pages/components/Footer.js [hunk 2]
```diff
@@ -3,52 +3,47 @@ import { makeStyles } from "@material-ui/core/styles";
...
-          <Link href="#" onClick={preventDefault}>
-            Media
-          </Link>
-        </div>
-        <Button variant="contained" onClick={rules}>
-          rules
-        </Button>
+      <Paper square>
+        <Typography className={classes.text}>
+           <b> Rules:</b>{rules}
```
Comment (bernardbaker) [inline react_front_end/src/pages/components/Footer.js:44]: You need ```{" "}``` before the ```</b>```.

File: react_front_end/src/pages/components/GameName.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ export const GameName = () => {
       className={classes.title}
     >
       <p data-testid="Game Name">
-        <span style={{ fontWeight: "bold" }}>Game</span>: {`${gameInfo.game}`}
+        <span style={{ fontWeight: "bold" }}>Game</span>: {`${gameInfo.name}`}
```
Comment (bernardbaker) [inline react_front_end/src/pages/components/GameName.js]: Change this back to ```{`${gameInfo.game}`}``` for the time being please.

File: react_front_end/src/pages/components/LandingPageInformation.js [hunk 4]
```diff
@@ -22,7 +31,41 @@ const LandingPageInformation = () => {
   return (
     <>
       <section className={classes.root}>
-        <ExperimentInformation />
+        <Paper square style={{maxWidth:"850px"}}>
```
Comment (bernardbaker) [inline react_front_end/src/pages/components/LandingPageInformation.js]: The copy needs to be split into sizeable chunks of information.

1) It’s Spring 2020 and Coachella, SXSW, the NBA, NHL, MLB and the Tokyo Olympics are cancelled this year.

2) We’re at the height of human technology and innovation, but at the same time facing one of the most devastating viral pandemics in history.

So it reads like a song, poem or story. I'm sure @un7c0rn can advise on how best to split it up into sizeable chunks.

There is a swipable library on NPM (https://www.npmjs.com/package/react-swipeable) use it to swipe the text left and right.

File: react_front_end/src/pages/components/TriangleLogo.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import Paper from "@material-ui/core/Paper";
+import Typography from "@material-ui/core/Typography";
+import logo from "../../assets/vir-us_triangle_logo_white.jpeg";
+import Grid from "@material-ui/core/Grid";
```
Comment (bernardbaker) [inline react_front_end/src/pages/components/TriangleLogo.js]: Don't use the grid here as a solution to not using flex-box. We should pair up on this.

**Review by bernardbaker — DISMISSED** [review]
@dzx11 

1) The image asset files; they are very big in dimensions and file size, is there a reason for this? The target demographic will all be using smart phones.

Consider using the HTML picture tag, which allows for source sets. Or reduce the dimensions, which will reduce the page load. Slow 3G took around 12 seconds to render the page.

2) The top-level HTML tag/body needs to have the background colour set to black.

3) This page doesn't show the game information http://localhost:8888/game-info/ebxglQBYU2QD6n2blRDH

4) http://localhost:8888/game-info/ebxglQBYU2QD6n2blRDH buttons are two wide. Reduce the width to maybe 80vw. And use ```flex-direction: column; justify-content: "center"; align-items: "center"``` on the ```form``` style object.

5) Start game page needs some work. Try to match the design as best possible. I think we should pair up on this. Flex box will allow you to match the design.

6) I think you should use SVG graphics where possible. We can ask @un7c0rn for SVG assets. SVG assets can be extracted from a PDF file.

7) Triangle logo should have stats on the left side and the right side. Flex-box will do this for you on both mobile and desktop.

I made a few changes to the CSS and Flex Box should solve a lot of the UI issues.

I'll pair up with you tomorrow morning if you can.

File: react_front_end/src/pages/components/Footer.js (hunk 2 above)
Comment (dzx11) [reply react_front_end/src/pages/components/Footer.js:44]: Not sure what you mean? If you mean the {rules} should be bold as well, that won't match the design.

File: react_front_end/src/pages/components/GameName.js (hunk 3 above)
Comment (dzx11) [reply react_front_end/src/pages/components/GameName.js]: Can you explain why? gamInfo.game does not return anything in my testing.

**Comment by dzx11** [comment]
@un7c0rn Can you point me to the updated logo? If it's the vir-us_logotype_new.png file, this one won't work with a black background because the letters would be black on black.

**Comment by bernardbaker** [comment]
Use a PDF file import data into a vector editor and change the colours.

That should work.

**Comment by bernardbaker** [comment]
David,

Here you go... let me know if you need anything else.


Regards,

Bernard T. A. Baker

**Comment by dzx11** [comment]
@bernardbaker I think that upload didn't work. Do you know any free tools that would do this?

**Comment by bernardbaker** [comment]
I'll try again.

This is what I see when I open the file in Chrome. But I'm not saying
you're wrong. I'll send them again.

[image: Screenshot 2020-04-27 at 17.20.39.png]


Regards,

Bernard T. A. Baker

**Comment by bernardbaker** [comment]
vir-us.logo.type.white.letters.svg
<https://drive.google.com/file/d/1_e00CfMEZDCGqRhNaygTmqNiKxN4oGNi/view?usp=drive_web>
[image: vir-us.logo.type.white.letters.jpg]
[image: vir-us.logo.type.white.letters.png]



Regards,

Bernard T. A. Baker

**Comment by dzx11** [comment]
@bernardbaker The drive link worked, thanks!

**Comment by bernardbaker** [comment]
Ok

**Comment by dzx11** [comment]
@un7c0rn @bernardbaker I just pushed my latest changes. 
1. The SVG map is hard to see because of the low contrast, and scaling it is difficult because the aspect ratio needs to remain the same as the source. Any ideas on how to improve the appearance?
2. @bernardbaker In the triangleLogo.js file I rely on in-line styles in order to do conditional styles. I rely on the state of props.hideTriangle to set margins and other styles in-line. This looks pretty janky in the code - is there a better way to do this?

**Comment by dzx11** [comment]
Also thanks @bernardbaker for the work on the triangle logo flexbox and converting the PDF files to SVG!

**Comment by bernardbaker** [comment]
No problem, you're more than welcome.

Regards,

Bernard T. A. Baker

**Comment by dzx11** [comment]
🐛look at ipad pro
🐛overflow bug on start game page
🐛! game info page: horizontal view (Galaxy s5, iphone 5se, iphone 4)

**Comment by bernardbaker** [comment]
🐛
Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering.
    in GameInfo (at GameInfoPage.js:27)
    in section (at GameInfoPage.js:25)
    in GameInfoPage (at App.js:80)
    in Route (at App.js:80)
    in Routes (at App.js:77)
    in Router (created by BrowserRouter)
    in BrowserRouter (at App.js:76)
    in ThemeProvider (at App.js:75)
    in Suspense (at App.js:74)
    in ErrorBoundary (created by UseErrorBoundaryWrapper)
    in UseErrorBoundaryWrapper (at App.js:64)
    in App (at src/index.js:11)

@dzx11 could you provide instructions on how to reproduce this bug?

@dzx11 Could you google the warning **Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering.**

**Comment by dzx11** [comment]
This looks great, thanks for your help.
@un7c0rn Ready for review
My only concern is that the query selector code might make the code brittle in the future. The difference between the latest commit and the previous one is that the input fields have square corners instead of round corners, and the labels as well as the input are centered. 

edit: My mistake - the difference is solely that the difference in the input/label styling. The previous commit did the square corners.

**Comment by bernardbaker** [comment]
The query selectors used developer defined IDs. I don't think we'll need to
change the IDs. So this shouldn't be a problem.

**Comment by bernardbaker** [comment]
**fixed** 🐛! game info page: horizontal view (Galaxy s5, iphone 5se, iphone 4)

**Comment by dzx11** [comment]
@bernardbaker the verification is no longer working. I.E. the textbox no longer turns red when submit button is clicked with invalid input. Was that intended? Also I'm seeing some artifacts that weren't there before around the textboxes that show up when entering input.

**Comment by bernardbaker** [comment]
What browser are you using? I missed the validation bug.

Do you want to pair up, or can you fix it?

**Comment by bernardbaker** [comment]
I took a look at the bug. It's the custom placement overriding the form
defaults.

I'll get this sorted tomorrow morning 👍.

**Pull request merged**